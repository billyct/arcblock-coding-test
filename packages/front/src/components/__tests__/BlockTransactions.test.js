import {render} from '@testing-library/react'

jest.mock('../../hooks/useBlock')
jest.mock('../../hooks/useTransactions')
const useBlock = require('../../hooks/useBlock')
const useTransactions = require('../../hooks/useTransactions')

jest.mock('../Card', () => 'card-component')
jest.mock('../Loading', () => 'loading-component')
jest.mock('../Paginate', () => 'paginate-component')
jest.mock('../TransactionItem', () => 'transaction-item-component')

const BlockTransactions = require('../BlockTransactions').default

afterAll(() => {
  jest.resetModules()
})

describe.each([
  [
    {block: undefined},
    {transactions: []},
  ],
  [
    {block: {transactionsCount: 2000}},
    {transactions: undefined},
  ],
  [
    {block: {transactionsCount: 2000}},
    {
      transactions: [
        {hash: 'hash 1'},
        {hash: 'hash 2'},
      ]
    },
  ]
])('test BlockTransactions Component with useBlock(%o) useTransactions(%o)', (block, transactions) => {
  beforeAll(() => {
    useBlock.__setMockData(block)
    useTransactions.__setMockData(transactions)
  })

  it('should render correct', function () {
    const {container} = render(<BlockTransactions/>)
    expect(container).toMatchSnapshot()
  })
})
