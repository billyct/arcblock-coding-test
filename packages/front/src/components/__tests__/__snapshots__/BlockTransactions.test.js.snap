// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test BlockTransactions Component with useBlock({ block: { transactionsCount: 2000 } }) useTransactions({
  transactions: [ { hash: 'hash 1' }, { hash: 'hash 2' }, [length]: 2 ]
}) should render correct 1`] = `
<div>
  <card-component
    classname="-mx-6"
    title="Transactions (2000)"
  >
    <transaction-item-component
      transaction="[object Object]"
    />
    <transaction-item-component
      transaction="[object Object]"
    />
    <paginate-component />
  </card-component>
</div>
`;

exports[`test BlockTransactions Component with useBlock({ block: { transactionsCount: 2000 } }) useTransactions({ transactions: undefined }) should render correct 1`] = `
<div>
  <card-component
    classname="-mx-6"
    title="Transactions (2000)"
  >
    <loading-component />
    <paginate-component />
  </card-component>
</div>
`;

exports[`test BlockTransactions Component with useBlock({ block: undefined }) useTransactions({ transactions: [ [length]: 0 ] }) should render correct 1`] = `<div />`;
